inc/backend_class.hpp

```diff
@@ -0,0 +1,25 @@
+//  For licensing see accompanying LICENSE file.
+//  Copyright © 2024 Argmax, Inc. All rights reserved.
+#pragma once 
+
+#if defined(QNN_DELEGATE)
+#include <tflite_qnn_model.hpp>
+#define MODEL_SUPER_CLASS     TFLiteQNN
```

Comment:
followup change:
singleton factory, produce delegate backend instances and interfaces with compiler flags for a given target.  use ABC + inheritance to override methods.

Please look at how delegates are implemented within e.g. ONNX, TFLite etc. and how they structure their classes in a similar way